### PR TITLE
ci: Upload integration tests coverage report to Codecov

### PR DESCRIPTION
**Pull request by siteshwar**
Related: https://gitlab.cee.redhat.com/covscan/covscan/-/issues/264

### CODE REVIEW

**Comment by siteshwar** [comment]
Codecov merges coverage for both unit and integration tests into one. Shall we separate them or keep it as it is?

**Comment by lzaoral** [comment]
I'd keep as it is.

File: containers/scripts/generate_integration_test_coverage.sh [hunk 1]
```diff
@@ -132,8 +132,15 @@ main() {
...
-    podman exec -it osh-client /usr/bin/coverage-3.6 html
+    # Avoid generating html reports in GitHub Actions CI
+    if [[ "$GITHUB_ACTIONS" != "true" ]];
+    then
+        # Convert test coverage to html
+        podman exec -it osh-client /usr/bin/coverage-3.6 html
+    else
+        # We use codecov in GitHub Actions CI. Upload xml reports to it.
+        podman exec -it osh-client /usr/bin/coverage-3.6 xml
+    fi
```
Comment (lzaoral) [inline containers/scripts/generate_integration_test_coverage.sh:143]: nit: I'd test for quality instead with swapped `then` and `else` branches.

**Review by lzaoral — APPROVED** [review]
LGTM! Thank you @siteshwar!

**Pull request merged**